app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/EnterMovementReferenceNumberController.scala

```diff
@@ -54,514 +49,118 @@ class EnterMovementReferenceNumberController @Inject() (
-
-                val updatedJourney = fillingOutClaim.copy(draftClaim = newDraftClaim)
-
-                val updateSessionWithReference = EitherT
-                  .liftF(updateSession(sessionStore, request)(_.copy(journeyStatus = Some(updatedJourney))))
-                  .leftMap((_: Unit) => Error("could not update session"))
-
-                val getDeclaration: EitherT[Future, Error, Option[DisplayDeclaration]] = claimService
-                  .getDisplayDeclaration(mrn)
-                  .leftMap(_ => Error("could not get declaration"))
-
-                val result: EitherT[Future, Error, Either[MrnImporter, ThirdPartyImporter]] = for {
-                  maybeDisplayDeclaration <- getDeclaration
-                  _                       <- updateSessionWithReference
-                  mrnJourneyFlow          <-
-                    EitherT
-                      .fromEither[Future](
-                        evaluateMrnJourneyFlow(fillingOutClaim.signedInUserDetails, maybeDisplayDeclaration)
-                      )
-                      .leftMap(_ => Error("could not evaluate MRN flow"))
-                  displayDeclaration      <-
-                    EitherT.fromOption[Future](maybeDisplayDeclaration, Error("could not unbox display declaration"))
-                  _                       <- EitherT.liftF(
-                                               updateSession(sessionStore, request)(
-                                                 _.copy(journeyStatus =
-                                                   Some(
-                                                     fillingOutClaim.copy(draftClaim =
-                                                       newDraftClaim.copy(
-                                                         displayDeclaration = Some(displayDeclaration)
-                                                       )
-                                                     )
-                                                   )
-                                                 )
-                                               )
-                                             )
-                } yield mrnJourneyFlow
+  def enterMrnSubmit(): Action[AnyContent]  = mrnSubmit(false)
+  def changeMrnSubmit(): Action[AnyContent] = mrnSubmit(true)
 
-                result.fold(
-                  e => {
-                    logger.warn("could not get declaration information", e)
-                    Redirect(baseRoutes.IneligibleController.ineligible())
-                  },
-                  {
-                    case Left(_)  => Redirect(routes.CheckDeclarationDetailsController.checkDetails())
-                    case Right(_) => Redirect(routes.EnterImporterEoriNumberController.enterImporterEoriNumber())
-                  }
-                )
-            }
-          )
-      }
-    }
-
-  def changeMrnSubmit(): Action[AnyContent] =
+  def mrnSubmit(isCYA: Boolean): Action[AnyContent] =
```

Comment:
I think changing this variable from isAmend to isCYA should be discussed more widely with the team before you do it.

Reply:
Reverted